app/static/tests/e2e/sessions.etest.ts

```diff
@@ -270,4 +284,38 @@ test.describe("Sessions tests", () => {
         await page.click("#load-session-from-code");
         await expect(await page.url()).toBe("http://localhost:3000/apps/day1/?share=good-dog");
     });
+
+    test("session initialise modal behaves as expected", async () => {
+        const browser = await usePersistentContext();
+        const page = await browser.newPage();
+        await loadAppPage(page);
+
+        // We don't see the modal on load first session...
+        await expect(await page.locator("#session-initialise-modal .modal")).not.toBeVisible({ timeout });
+
+        // ..but we should on the second
+        await loadAppPage(page);
+        await expect(await page.locator("#session-initialise-modal .modal")).toBeVisible({ timeout });
+
+        // select new session and make some changes, including set label
+        await page.click("#new-session");
+        await expect(await page.locator("#session-initialise-modal .modal")).not.toBeVisible({ timeout });
+
+        await page.click(":nth-match(.wodin-left .nav-tabs a, 2)"); // select code tab
+        await writeCode(page, newFitCode);
+        await page.click("#sessions-menu");
+        await page.click("#edit-current-session-label");
+        await enterSessionLabel(page, "header-edit-session-label", "session to reload");
+        await page.waitForTimeout(saveSessionTimeout);
+
+        // refresh page and select reload latest - should see the changes and new label
+        await loadAppPage(page);
+        await expect(await page.locator("#session-initialise-modal .modal")).toBeVisible({ timeout });
+        await page.click("#reload-session");
+        await page.click(":nth-match(.wodin-left .nav-tabs a, 2)"); // select code tab
```

Comment:
is there not a nicer Locators we can use to target these elements in tests? i.e the build in ones, such as `getByRole`, `getBytext` etc

Reply:
I'm not a big fan of nth matching either actually, but my idea of nice locators are ids and classes, as they feel a bit more stable than text (and maybe role as that can be ambiguous) which may change. I think it's fine to check expected text in a single test but I'd tend to use id or class to identify elements as more semantically reliable. But you're right that this means your test is a little divorced from the user experience... Let's maybe have a group chat with the other about this sometime?

Reply:
yeah but thats agains the reccomended way to test the UI. Its best to test the UI for what the users can see.. thus best is `getByRole`.. The user can not see ids and classes

Reply:
Yup, fair point. As discussed, let's leave these as they are, consistent with the existing tests, and try out some refactoring of e2e tests to use roles in a separate ticket.